practice-app/api/views.py

```diff
@@ -18,10 +21,72 @@ def post(self, request):
             'message': f'Hello, {name}.'
         })
 
+
 class LoginAPIView(APIView):
     def post(self, *args, **kwargs):
         data = self.request.data
         serializer = ls.LoginSerializer(data=data)
         serializer.is_valid(raise_exception=True)
         return Response(serializer.data, status=st.HTTP_200_OK)
 
+
+class TotalProfitAPIView(APIView):
+    authentication_classes = (JSONWebTokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    # Helper function for calculating total profit with respect to given symbol.
+    # Default is Turkish Liras
+
+    def get_total_profit(self, username, symbol="TRY"):
+        user = User.objects.filter(username=username)[0]
+
+        # Getting investments of a particular user
+
+        investments = ManualInvestment.objects.filter(made_by=user)
+        default_eq = Equipment.objects.filter(symbol=symbol)[0]
+        profit = 0
+
+        for investment in investments:
+
+            base_eq = investment.base_equipment
+            target_eq = investment.target_equipment
+            base_amount = investment.base_amount
+            target_amount = investment.target_amount
+
+            # If the sold equipment is the same as default eqipment, ratio is 1.
+            if base_eq.symbol == default_eq.symbol:
```

Comment:
You may delete the if/else statement because we are going to have e.g. USD:USD parity whose ratio is 1 in our database. CronJob queries handle this situation.